src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -765,7 +767,10 @@ private IDynamicInvocationOperation CreateBoundDynamicCollectionElementInitializ
             ITypeSymbol type = boundCollectionElementInitializer.Type;
             Optional<object> constantValue = ConvertToOptional(boundCollectionElementInitializer.ConstantValue);
             bool isImplicit = boundCollectionElementInitializer.WasCompilerGenerated;
-            return new LazyDynamicInvocationExpression(operation, arguments, argumentNames: ImmutableArray<string>.Empty, argumentRefKinds: ImmutableArray<RefKind>.Empty, _semanticModel, syntax, type, constantValue, isImplicit);
+            BoundImplicitReceiver implicitReceiver = boundCollectionElementInitializer.ImplicitReceiver;
+            Lazy<IOperation> addReference = new Lazy<IOperation>(() =>
+                CreateBoundDynamicMemberAccessOperation(implicitReceiver, typeArgumentsOpt: ImmutableArray<TypeSymbol>.Empty, memberName: "Add", implicitReceiver.Syntax, type: null, value: default, isImplicit: true));
```

Comment:
This is a very long line. Consider splitting it.

Reply:
Will do in a follow up. Doesn't look long on a 4K monitor :).

---
In reply to: [192826248](https://github.com/dotnet/roslyn/pull/27308#discussion_r192826248) [](ancestors = 192826248)